deploy/script/src/index.js

```diff
@@ -0,0 +1,80 @@
+const fs = require('fs');
+
+const config = require('./config');
+const { execCommand } = require('./util');
+const List = require('prompt-list');
+
+const askServiceToDeploy = async () => {
+  const choices = Object.keys(config.deploy);
+  let serviceToDeploy;
+
+  if (config.service) {
+    serviceToDeploy = config.service;
+
+    if (!choices.includes(serviceToDeploy)) {
+      throw new Error(`Wrong service specified to deploy [${serviceToDeploy}]. Aborting`);
+    }
+  } else {
+    const list = new List({
+      name: 'service',
+      message: 'What service to deploy?',
+      choices,
+    });
+
+    serviceToDeploy = await list.run();
+  }
+
+  const serviceConfig = config.deploy[serviceToDeploy];
+  serviceConfig.name = serviceToDeploy;
+
+  return serviceConfig;
+};
+
+// const dockerIgnore = `
+// node_modules
+// `;
+
+const buildAndPushImage = async ({ dockerFilePath, dockerRepo, dockerContextDir, imageTag }) => {
+  await execCommand(`docker build -f ${dockerFilePath} -t ${dockerRepo} ${dockerContextDir}`);
+  await execCommand(`docker tag ${dockerRepo} ${imageTag}`);
+  await execCommand(`docker push ${imageTag}`);
+}
+
+const pushToKubernetes = async ({ imageTag, appName }) => {
+  const deployDir = `${config.rootDir}/deploy/app`;
+
+  if (config.kubeConfig) {
+    console.log('Creating kubeconfig');
+    fs.mkdirSync(`${config.home}/.kube`);
+    fs.writeFileSync(`${config.home}/.kube/config`, config.kubeConfig);
+  }
+
+  await execCommand(`
+    helm upgrade --install apps-${appName} ${deployDir} --namespace app \
+      --set appname=${appName} \
+      --set imagesVersion=${imageTag} \
+      -f ${deployDir}/values/values.yml \
+  `);
+}
+
+const deploy = async () => {
+  if (config.dockerRegistry.password) {
+    await execCommand(`docker login --username ${config.dockerRegistry.username} --password ${config.dockerRegistry.password}`);
```

Comment:
if we start using js scripts, would be useful to have an abstraction layer over `execCommand` to make script looks more friendly (like `dockerLogin(username, password)`) or even use an sdk https://github.com/apocas/dockerode

Reply:
And pretty the same can be said to any `execCommand` call here

Reply:
For now it's simpler to keep raw `execCommand` to avoid overcomplications and learning additional DSLs. execCommand can be used to execute any shell command needed for the project build and gives more flexibility, teams can decide on more verbose solutions on project level if needed